docker/main/ngen-calibration/entrypoint.sh

```diff
@@ -1,50 +1,94 @@
-#!/bin/sh
-# Managed by the _generate_docker_cmd_args function in scheduler.py of dmod.scheduler
-#
-# $1 will have the number of nodes associated with this run
-# $2 will have comma-delimited host strings in MPI form; e.g., hostname:N,hostname:M
-# $3 will have the unique job id
-# $4 is the worker index
-# $5 will be the name of the output dataset (which will imply a directory location)
-# $6 will be the name of the hydrofabric dataset (which will imply a directory location)
-# $7 will be the name of the realization configuration dataset (which will imply a directory location)
-# $8 will be the name of the BMI configuration dataset (which will imply a directory location)
-# $9 will be the name of the partition configuration dataset (which will imply a directory location)
-# TODO: wire up $10
-# $10 will be the name of the calibration configuration dataset (which will imply a directory location)
-
-# Not yet supported
-# no-op
-MPI_NODE_COUNT="${1:?No MPI node count given}"
-# no-op
-MPI_HOST_STRING="${2:?No MPI host string given}"
-# no-op
-PARTITION_DATASET_NAME="${9:?}"
-
-JOB_ID=${3:?No Job id given}
-WORKER_INDEX=${4:?No worker index given}
-
-OUTPUT_DATASET_NAME="${5:?}"
-HYDROFABRIC_DATASET_NAME="${6:?}"
-REALIZATION_CONFIG_DATASET_NAME="${7:?}"
-BMI_CONFIG_DATASET_NAME="${8:?}"
-CALIBRATION_CONFIG_DATASET_NAME="${10:?}"
+#!/bin/bash
+# Args are managed by the _generate_docker_cmd_args function in scheduler.py of dmod.scheduler
 
+# TODO: Docker secret variable values need to be parameterized
 ACCESS_KEY_SECRET="object_store_exec_user_name"
 SECRET_KEY_SECRET="object_store_exec_user_passwd"
 DOCKER_SECRETS_DIR="/run/secrets"
 ACCESS_KEY_FILE="${DOCKER_SECRETS_DIR}/${ACCESS_KEY_SECRET}"
 SECRET_KEY_FILE="${DOCKER_SECRETS_DIR}/${SECRET_KEY_SECRET}"
 
+MPI_USER="mpi"
+if [ "$(whoami)" = "${MPI_USER}" ]; then
+    MPI_HOSTS_FILE="${HOME}/.mpi_hosts"
+else
+    MPI_HOSTS_FILE="$(su ${MPI_USER} -c 'echo "${HOME}"')/.mpi_hosts"
+fi
+RUN_SENTINEL="/home/${MPI_USER}/.run_sentinel"
```

Comment:
Hmm. To go that route, I'd want us to add a dedicated directory under `/var` as part of the image.  It would hold job-execution-items like this and be `chown`-ed to `mpiuser`.  Otherwise we eventually won't be able to create the sentinel file in the container (see #238).

We also aren't really writing anything to this file, just creating, watching, and removing it.  We can discuss further, but I'm inclined to leave it as-is for now.

Reply:
Im totally okay with leave it how it is. The only reason I asked was in the situation that an `MPI_USER` did not have a `/home` directory.